File: samples/mapzen-sample/src/main/java/mapzen/com/sdksampleapp/models/SampleMap.java
package mapzen.com.sdksampleapp.models;

import android.util.SparseArray;

import mapzen.com.sdksampleapp.R;
import static mapzen.com.sdksampleapp.models.MapSampleList.MAP_SAMPLES;

/**
 * Maps navigation ids to their list of {@link Sample}s.
 */
public class SampleMap {

  private SampleMap() { }

  public static final SparseArray<Sample[]> NAV_ID_TO_SECTIONS;
  static {
    NAV_ID_TO_SECTIONS = new SparseArray<>(4);
    NAV_ID_TO_SECTIONS.append(R.id.navigation_map, MAP_SAMPLES);
    NAV_ID_TO_SECTIONS.append(R.id.navigation_route, null);
    NAV_ID_TO_SECTIONS.append(R.id.navigation_search, null);
    NAV_ID_TO_SECTIONS.append(R.id.navigation_more, null);
  }

}

File: samples/mapzen-sample/src/main/java/mapzen/com/sdksampleapp/models/SampleMap.kt
package mapzen.com.sdksampleapp.models

import android.support.v4.util.SparseArrayCompat

import mapzen.com.sdksampleapp.R
import mapzen.com.sdksampleapp.models.MapSampleList.Companion.MAP_SAMPLES
import mapzen.com.sdksampleapp.models.MoreSampleList.Companion.MORE_SAMPLES
import mapzen.com.sdksampleapp.models.RouteSampleList.Companion.ROUTE_SAMPLES
import mapzen.com.sdksampleapp.models.SearchSampleList.Companion.SEARCH_SAMPLES

/**
 * Maps navigation ids to their list of [Sample]s.
 */
class SampleMap : SampleVendor {

  companion object {
    @JvmStatic val NAV_ID_TO_SECTIONS: SparseArrayCompat<Array<Sample>?> by lazy {
      val sparseArray = SparseArrayCompat<Array<Sample>?>(4)
      sparseArray.append(R.id.navigation_map, MAP_SAMPLES)
      sparseArray.append(R.id.navigation_route, ROUTE_SAMPLES)
      sparseArray.append(R.id.navigation_search, SEARCH_SAMPLES)
      sparseArray.append(R.id.navigation_more, MORE_SAMPLES)
      sparseArray
    }
  }

  override fun samplesForNavId(navId: Int): Array<Sample>? {
    return NAV_ID_TO_SECTIONS.get(navId)
  }
}
